CIP-0118/README.md

```diff
@@ -0,0 +1,1042 @@
+transactions in a zone once a single transaction is found to be phase-2 invalid.
+Recall that  
+it is not possible to exclude only _some_ transactions from a zone in a general way,
+so, unlike the existing design, the mempool is no able to reject individual zone transactions
+dependent on a preceding transaction being phase-2 valid. We design the `ZONE-N`
+rule to deal with this situation in a special way.
+
+The following checks are done by both rules :
+
+1. the sum total of the sizes of all transactions in the zone is less than the `maxTxSize`
+protocol parameter (see [Network Requirements and Changes](#network-requirements-and-changes)), and
+2. the collateral provided by each transaction `tx` for script execution is sufficient
+to cover **all scripts in all transactions preceding `tx` in the zone, including `tx` itself**
+(see [DDoS](#ddos))
+3. the `LEDGERS` transition is valid for the zone environment, transaction list,
+and initial `LStateTemp`, which is the initial `LState` of `ZONE` transition with the
+`FRxO` set to empty in `UTxOTemp`
+4. all collateral inputs must be in the initial `LState` UTxO set (see [DDoS](#ddos))
+
+**`ZONE-V`**. This rule has additional checks :
+
+1. there are no cycles in the transaction dependency graph (see [Flash Loans](#flash-loans))
+2. all `requiredTxs` for all transactions in the zone are present
+
+The updated state in this rule is computed by applying `LEDGERS` to the starting state,
+environment, and list of transactions in the zone.
+
+**`ZONE-N`**. This rule has additional checks :
+
+1. the only transaction that is phase-2 invalid in the zone is the last transaction
+
+The updated state in this rule is not the state computed by applying `LEDGERS`.
+We discard that state, and instead specify the updated `LState` directly.
+The only fields updated are :
+
+1. the collateral inputs are removed from the UTxO (and change returned)
+2. `fees` is updated to `fees` + the collateral amount in the last transaction of the zone
+
+#### UTXO
+
+The `UTXO` rule is updated to additionally check that the `fulfills`
+of the transaction being processed are present in the `FRxO` set.
+The `UTXO` rule checks the preservation of value (POV), which compares the `produced`
+and `consumed` value for the given transaction. The POV calculation is updated by
+adding
+
+1. the sum total value in the transaction `fulfills` to the `produced` value
+2. the sum total value in the transaction `requests` to the `consumed` value
+
+
+#### UTXOS
+
+The `UTXOS` rule is the one that actually computes the updates to
+the UTxO state done by a transaction. The UTxO set update remains the same
+as in the existing design. The `FRxO` is updated by removing
+all entries that are indexed by the values in the `fulfills` of the
+transaction, then adding all entries of the form `(txID, ix) ↦ r`, where
+`txID` is the ID of the transaction being processed, and `ix ↦ r` is
+in the `requests` of the transaction.
+
+## System Component Changes
```

Comment:
```suggestion
### System Component Changes
```

Reply:
Similar justification for this as for https://github.com/cardano-foundation/CIPs/pull/862#discussion_r1688272866.